### PR TITLE
[DebugInfo] Add a free helper to get empty locations, NFC

### PR DESCRIPTION
**Pull request by vedantk**
It seems more convenient to have a single function to get empty locations, instead of having to work out the connection between SourceLoc and RegularLocation.

@adrian-prantl @dcci thoughts?

### CODE REVIEW

**Comment by vedantk** [comment]
@swift-ci Please smoke test

**Comment by vedantk** [comment]
@swift-ci Please smoke test

**Comment by vedantk** [comment]
@swift-ci Please smoke test Linux platform

**Review by dcci — APPROVED** [review]
This seems plausible to me. Adrian?

**Comment by vedantk** [comment]
@swift-ci Please smoke test Linux platform

File: include/swift/SIL/SILLocation.h [hunk 1]
```diff
@@ -502,6 +502,10 @@ class RegularLocation : public SILLocation {
   }
 };
 
+/// Empty locations may be applied to instructions without any clear
+/// correspondence to an AST node.
+static inline RegularLocation getEmptyLocation() { return {SourceLoc()}; }
```
[the remark this answers is not in the document]
Comment (vedantk) [reply include/swift/SIL/SILLocation.h]: Works for me.

**Comment by vedantk** [comment]
@swift-ci Please smoke test

**Comment by vedantk** [comment]
@adrian-prantl Is this good to merge now?

File: lib/SILOptimizer/Transforms/FunctionSignatureOpts.cpp [hunk 2]
```diff
@@ -979,11 +979,11 @@ static void createArgumentRelease(SILBuilder &Builder, ArgumentDescriptor &AD) {
   if (Arg->getType().isAddress()) {
     assert(AD.PInfo->getConvention() == ParameterConvention::Indirect_In
            && F.getConventions().useLoweredAddresses());
-    Builder.createDestroyAddr(RegularLocation(SourceLoc()),
+    Builder.createDestroyAddr(getCompilerGeneratedLocation(),
```
Comment (adrian-prantl) [inline lib/SILOptimizer/Transforms/FunctionSignatureOpts.cpp]: This should possibly be a SILBuilderWithScope, but that's for another commit. Your change is NFC.

File: lib/SILOptimizer/Utils/Local.cpp [hunk 3]
```diff
@@ -50,7 +50,7 @@ swift::createIncrementBefore(SILValue Ptr, SILInstruction *InsertPt) {
 
   // Set up the builder we use to insert at our insertion point.
   SILBuilder B(InsertPt);
-  auto Loc = RegularLocation(SourceLoc());
+  auto Loc = getCompilerGeneratedLocation();
```
Comment (adrian-prantl) [inline lib/SILOptimizer/Utils/Local.cpp]: again, probably better a SILBuilderWithScope, but not in this patch.

**Review by adrian-prantl — APPROVED** [review]
Yes thanks, this is good!

**Comment by vedantk** [comment]
Thanks! I'll look into your comments about using SILBuilderWithScope.

**Pull request merged**